### PR TITLE
Add try catch to make sure all handlers are closed

### PR DESCRIPTION
**Pull request by ChristopheDuong**
## What
Related to hanging reset jobs as reported in airbytehq/alpha-beta-issues#7

## How
We have a common pattern of a consumer in destination connectors that wraps around multiple handlers/writers for each stream and tries to close them in a loop.

If one `handler.close()` throws an exception, it might interrupt and skip closing other threads/handlers leaving the connector in a stuck state.

In this PR, I prototype this first solution specific for S3Consumer/GcsConsumer to verify if this hypothesis is really the cause of jobs getting stuck on reset.
Then we can expand the solution in a more generic way to other consumers afterward once it's confirmed.

### CODE REVIEW

**Review by subodh1810 — APPROVED** [review]

File: airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/S3Consumer.java [hunk 1]
```diff
@@ -85,8 +90,20 @@ protected void acceptTracked(final AirbyteMessage airbyteMessage) throws Excepti
 
   @Override
   protected void close(final boolean hasFailed) throws Exception {
-    for (final DestinationFileWriter handler : streamNameAndNamespaceToWriters.values()) {
-      handler.close(hasFailed);
+    Optional<Exception> exceptionsThrown = Optional.empty();
```
Comment (tuliren) [inline airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/S3Consumer.java:93]: Different handle closure may fail due to different reasons. This variable is only able to store one exception at most. Should it be a list, and when empty, throw one exception including all the exception messages? The trade-off is that the aggregated exception won't include the stack trace. But I think that throwing out the exception message is more important than knowing all the stack trace.

**Review by tuliren — APPROVED** [review]

**Review by jrhizor — APPROVED** [review]

**Comment by ChristopheDuong** [comment]
I managed to reproduce the failing scenario locally and it turns out this fix does not handle the hanging stuff yet so I'm still investigating

**Comment by ChristopheDuong** [comment]
/publish connector=connectors/destination-bigquery-denormalized

**Review by subodh1810 — APPROVED** [review]

**Pull request merged**